src/pages/RepoPage/FailedTestsTab/FailedTestsPage/SelectorSection/BranchSelector/BranchSelector.tsx

```diff
@@ -76,15 +76,20 @@ const BranchSelector = () => {
     )
   }
 
-  const sortedBranchList = overview?.defaultBranch
-    ? [
+  const sortedBranchList = useMemo(() => {
```

Comment:
adds a check for !branchList?.branches to clean things up a bit

And wraps the whole thing in a useMemo